### PR TITLE
Windows: implement getCorrectCasing

### PR DESCRIPTION
**Pull request by laszlocsomor**
Implement getCorrectCasing: a function that
normalizes a path and returns the case-correct
version of it.

See https://github.com/bazelbuild/bazel/issues/8799

### CODE REVIEW

File: src/main/java/com/google/devtools/build/lib/windows/jni/WindowsFileOperations.java [hunk 1]
```diff
@@ -228,4 +230,14 @@ public static boolean deletePath(String path) throws IOException {
         throw new IOException(String.format("Cannot delete path '%s': %s", path, error[0]));
     }
   }
+
```
Comment (meisterT) [inline src/main/java/com/google/devtools/build/lib/windows/jni/WindowsFileOperations.java]: Please a comment here that explains what "correct" refers to.

File: src/main/native/windows/file.cc [hunk 2]
```diff
@@ -781,5 +781,46 @@ bool GetCwd(std::wstring* result, DWORD* err_code) {
   }
 }
 
+std::wstring GetCorrectCasing(const std::wstring& abs_path, bool with_unc) {
```
Comment (meisterT) [inline src/main/native/windows/file.cc]: this looks like an absolute path is required - I don't see this requirement on the Java side. Do I miss something?

File: src/main/native/windows/file.cc [hunk 3]
```diff
@@ -781,5 +781,46 @@ bool GetCwd(std::wstring* result, DWORD* err_code) {
...
+  if (!HasDriveSpecifierPrefix(abs_path.c_str())) {
+    return L"";
+  }
+  std::wstring path = Normalize(abs_path);
+  std::unique_ptr<wchar_t[]> result(new wchar_t[4 + path.size() + 1]);
+  wcscpy(result.get(), L"\\\\?\\");
+  wcscpy(result.get() + 4, path.c_str());
+  result[4] = towupper(result[4]);
+  wchar_t* start = result.get() + 7;
```
Comment (meisterT) [inline src/main/native/windows/file.cc]: There is a lot of implicit knowledge in this code which makes it hard to read, e.g. where does the 7 come from?

Comment (meisterT) [reply src/main/native/windows/file.cc]: is this from UNC (if so make it conditional)?

File: src/main/native/windows/file.cc (hunk 2 above)
Comment (meisterT) [reply src/main/native/windows/file.cc]: currently the only caller passes with_unc=false, right? Do you have plans to change that?

File: src/main/java/com/google/devtools/build/lib/windows/jni/WindowsFileOperations.java (hunk 1 above)
Comment (laszlocsomor) [reply src/main/java/com/google/devtools/build/lib/windows/jni/WindowsFileOperations.java]: Done.

File: src/main/native/windows/file.cc (hunk 2 above)
Comment (laszlocsomor) [reply src/main/native/windows/file.cc]: Removed.

File: src/main/native/windows/file.cc (hunk 3 above)
Comment (laszlocsomor) [reply src/main/native/windows/file.cc]: Added comment.

**Comment by laszlocsomor** [comment]
Thanks for the quick and thorough review!

**Pull request closed without merging**